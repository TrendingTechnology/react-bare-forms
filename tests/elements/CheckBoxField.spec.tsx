import * as React from "react";
import renderer from "react-test-renderer";
import {
    Form, CheckBoxField, isChecked
} from "../../src";



describe("#CheckBoxField()", () => {
    it("should render an input field with the bootstrap styled tags", () => {
        let state = {
            terms: false,
        };

        let component = renderer.create(
            <Form state={state}>
                <CheckBoxField name="username" checked={state.terms} />
            </Form>
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render a single input tag with no bootstrap styles", () => {
        let state = {
            terms: false,
        };
        let component = renderer.create(
            <Form state={state}>
                <CheckBoxField name="username" checked={state.terms} />
            </Form>
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });


    it("it should pass in props", () => {
        let state = {
            terms: true,
        };

        let testFormRederer: any = renderer.create(
            <Form state={state}>
                <CheckBoxField name="username" checked={state.terms} />
            </Form>
        );

        const testFormInstance = testFormRederer.root;
        expect(testFormInstance.props.state.terms).toEqual(state.terms);

    });

    it("it should display validation errors", () => {
        let state = {
            terms: false,
        };

        let component = renderer.create(
            <Form state={state} name="myForm">
                <CheckBoxField
                    name="terms"
                    checked={state.terms}
                    validators={[isChecked()]}
                />
            </Form>
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("it should pass validation props", () => {
        let state = {
            terms: true,
        };


        let testFormRenderer: any = renderer.create(
            <Form state={state} name="myForm">
                <CheckBoxField
                    name="terms"
                    checked={state.terms}
                    validators={[isChecked()]}
                />
            </Form>
        );

        const testFormInstance = testFormRenderer.root;
        expect(testFormInstance.props.state.terms).toEqual(state.terms);
    });

});
