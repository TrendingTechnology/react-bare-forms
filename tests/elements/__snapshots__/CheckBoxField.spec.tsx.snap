// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#CheckBoxField() it should display validation errors 1`] = `
<form
  name="myForm"
  onSubmit={[Function]}
  state={
    Object {
      "terms": false,
    }
  }
>
  <div
    className="form-group form-check"
  >
    <input
      checked={false}
      className="form-check-input "
      name="terms"
      onChange={[Function]}
      type="checkbox"
    />
  </div>
</form>
`;

exports[`#CheckBoxField() should render a single input tag with no bootstrap styles 1`] = `
<form
  onSubmit={[Function]}
  state={
    Object {
      "terms": false,
    }
  }
>
  <div
    className="form-group form-check"
  >
    <input
      checked={false}
      className="form-check-input "
      name="username"
      onChange={[Function]}
      type="checkbox"
    />
  </div>
</form>
`;

exports[`#CheckBoxField() should render an input field with the bootstrap styled tags 1`] = `
<form
  onSubmit={[Function]}
  state={
    Object {
      "terms": false,
    }
  }
>
  <div
    className="form-group form-check"
  >
    <input
      checked={false}
      className="form-check-input "
      name="username"
      onChange={[Function]}
      type="checkbox"
    />
  </div>
</form>
`;
